ext/standard/string.c

```diff
@@ -1192,6 +1192,105 @@ PHP_FUNCTION(strtolower)
 }
 /* }}} */
 
+PHP_FUNCTION(str_increment)
+{
+	zend_string *str;
+
+	ZEND_PARSE_PARAMETERS_START(1, 1)
+		Z_PARAM_STR(str)
+	ZEND_PARSE_PARAMETERS_END();
+
+	if (ZSTR_LEN(str) == 0) {
+		zend_argument_value_error(1, "cannot be empty");
+		RETURN_THROWS();
+	}
+	if (!zend_string_only_has_ascii_alphanumeric(str)) {
+		zend_argument_value_error(1, "must be composed only of alphanumeric ASCII characters");
+		RETURN_THROWS();
+	}
+
+	zend_string *incremented = zend_string_init(ZSTR_VAL(str), ZSTR_LEN(str), /* persistant */ false);
+	size_t position = ZSTR_LEN(str)-1;
+	bool carry = false;
+
+	do {
+		char c = ZSTR_VAL(incremented)[position];
+		if (EXPECTED( (c >= 'a' && c < 'z') || (c >= 'A' && c < 'Z') || (c >= '0' && c < '9') )) {
```

Comment:
Could be simplified to `c != 'z' && c != 'Z' && c != '9'`, or `zend_string_only_has_ascii_alphanumeric` could be inlined to avoid a double iteration of the string.

Reply:
Indeed.